src/libraries/System.Text.Json/src/System/Text/Json/Reader/Utf8JsonReader.TryGet.cs

```diff
@@ -38,17 +38,169 @@ namespace System.Text.Json
+        /// Thrown if trying to get the value of the JSON token that is not a string
+        /// (i.e. other than <see cref="JsonTokenType.String"/> or <see cref="JsonTokenType.PropertyName"/>.
+        /// <seealso cref="TokenType" />
+        /// It will also throw when the JSON string contains invalid UTF-8 bytes, or invalid UTF-16 surrogates.
+        /// </exception>
+        /// <exception cref="ArgumentException">The destination buffer is too small to hold the unescaped value.</exception>
+        public readonly int CopyString(Span<byte> utf8Destination)
+        {
+            if (_tokenType is not (JsonTokenType.String or JsonTokenType.PropertyName))
+            {
+                ThrowHelper.ThrowInvalidOperationException_ExpectedString(_tokenType);
+            }
+
+            int bytesWritten;
+
+            if (ValueIsEscaped)
+            {
+                if (!TryCopyEscapedString(utf8Destination, out bytesWritten))
+                {
+                    utf8Destination.Slice(0, bytesWritten).Clear();
+                    ThrowHelper.ThrowArgumentException_DestinationTooShort();
+                }
+            }
+            else
+            {
+                if (HasValueSequence)
+                {
+                    ReadOnlySequence<byte> valueSequence = ValueSequence;
+                    valueSequence.CopyTo(utf8Destination);
+                    bytesWritten = (int)valueSequence.Length;
+                }
+                else
+                {
+                    ReadOnlySpan<byte> valueSpan = ValueSpan;
+                    valueSpan.CopyTo(utf8Destination);
+                    bytesWritten = valueSpan.Length;
+                }
+            }
+
+            JsonReaderHelper.ValidateUtf8(utf8Destination.Slice(0, bytesWritten));
+            return bytesWritten;
+        }
+
+        /// <summary>
+        /// Copies the current JSON token value from the source, unescaped, and transcoded as a UTF-16 char buffer.
+        /// </summary>
+        /// <param name="destination">A buffer to write the transcoded UTF-16 characters into.</param>
+        /// <returns>The number of characters written to <paramref name="destination"/>.</returns>
+        /// <remarks>Unlike <see cref="GetString"/>, this method does not support <see cref="JsonTokenType.Null"/>.</remarks>
+        /// <exception cref="InvalidOperationException">
+        /// Thrown if trying to get the value of the JSON token that is not a string
+        /// (i.e. other than <see cref="JsonTokenType.String"/> or <see cref="JsonTokenType.PropertyName"/>.
+        /// <seealso cref="TokenType" />
+        /// It will also throw when the JSON string contains invalid UTF-8 bytes, or invalid UTF-16 surrogates.
+        /// </exception>
+        /// <exception cref="ArgumentException">The destination buffer is too small to hold the unescaped value.</exception>
+        public readonly int CopyString(Span<char> destination)
```

Comment:
can existing APIs be simplified by using any of these new methods?

Reply:
If you mean converter implementations, yes, I've refactored a bunch of them in this PR.